packages/react-components/react-tree/src/hooks/useFlatTreeNavigation.ts

```diff
@@ -0,0 +1,103 @@
+import { ArrowDown, ArrowLeft, ArrowRight, ArrowUp, End, Home } from '@fluentui/keyboard-keys';
+import { useFluent_unstable } from '@fluentui/react-shared-contexts';
+import { isHTMLElement } from '@fluentui/react-utilities';
+import { TreeNavigationData_unstable } from '../Tree';
+import { filterTreeItem } from '../utils/filterTreeItemAndGroup';
+import { isHTMLElementWithRole } from '../utils/isHTMLElementWithRole';
+import { useDOMTreeWalker } from './useDOMTreeWalker';
+import type { TreeItemPropsReferences } from './useFlatTreeItems';
+import { focusTypeAhead } from './useNestedTreeNavigation';
+
+export function useFlatTreeNavigation(references: TreeItemPropsReferences) {
+  const { targetDocument } = useFluent_unstable();
+  const [treeWalkerRef, rootRef] = useDOMTreeWalker(NodeFilter.SHOW_ELEMENT, {
+    acceptNode: filterTreeItem,
+  });
+
+  function navigate(data: TreeNavigationData_unstable) {
+    const { current: treeWalker } = treeWalkerRef;
+    if (!treeWalker || !targetDocument) {
+      return;
+    }
+    switch (data.type) {
+      case ArrowLeft:
+        return focusSubtreeOwnerItem(data.target, { references, document: targetDocument });
+      case ArrowRight:
+        return focusFirstSubtreeItem(data.target, { treeWalker });
+      case 'TypeAhead':
+        return focusTypeAhead(data.target, { treeWalker, key: data.event.key });
+      case End:
+        return focusLastTreeItem({ treeWalker });
+      case Home:
+        return focusFirstTreeItem({ treeWalker });
+      case ArrowDown:
+        return focusNextTreeItem(data.target, { treeWalker });
+      case ArrowUp:
+        return focusPreviousTreeItem(data.target, { treeWalker });
+    }
+  }
+  return [navigate, rootRef] as const;
+}
+
+type TreeWalkerOptions = { treeWalker: TreeWalker };
+
+function focusFirstSubtreeItem(target: HTMLElement, { treeWalker }: TreeWalkerOptions) {
+  treeWalker.currentNode = target;
+  const nextNode = treeWalker.nextNode();
+  if (!isHTMLElementWithRole(nextNode, 'treeitem')) {
```

Comment:
if someone wraps their treeitem with a layout wrapper, this will break right? because you only check if the next node has the treeitem role

```tsx
<treeitem />
<div role="presentation>
  <treeitem />
</div>
```

Layout divs would happen quite often with virtualizatio